devtools/maven/src/main/java/io/quarkus/maven/CreateProjectMojo.java

```diff
@@ -160,18 +163,52 @@ public void execute() throws MojoExecutionException {
         } catch (IOException e) {
             throw new MojoExecutionException("Could not create directory " + outputDirectory, e);
         }
+
         File projectRoot = outputDirectory;
         File pom = new File(projectRoot, "pom.xml");
+        Model parentPomModel = null;
```

Comment:
Here is one thing I noticed which is not necessarily related to your changes but may help anyway.
Checking whether we are in the context of an existing Maven project could be done with `project.getFile() != null`, if it's true, the Maven project exists. And that would be the `pom` above. Which will also support `-f custom.xml` use-case. Would you like to re-write it using that?
Another class that may be helpful is `io.quarkus.bootstrap.utils.BuildToolHelper` it includes `isGradleProject(Path project)` method, although it goes all the way up in the FS tree, so, perhaps not the best choice here.
Otherwise, it looks good to me. Thanks!